src/mesh/mesh.hpp

```diff
@@ -273,74 +288,41 @@ class Mesh {
   bool GatherCostListAndCheckBalance();
   void RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput *app_in,
                                        int ntot);
-
+  void BuildGMGHierarchy(int nbs, ParameterInput *pin, ApplicationInput *app_in);
+  void SetSameLevelNeighbors(BlockList_t &block_list, const LogicalLocMap_t &loc_map,
+                             RootGridInfo root_grid, int nbs, bool gmg_neighbors,
+                             int composite_logical_level = 0);
   // defined in either the prob file or default_pgen.cpp in ../pgen/
   static void InitUserMeshDataDefault(Mesh *mesh, ParameterInput *pin);
   std::function<void(Mesh *, ParameterInput *)> InitUserMeshData =
       InitUserMeshDataDefault;
 
   void EnrollBndryFncts_(ApplicationInput *app_in);
-  void EnrollUserMeshGenerator(CoordinateDirection dir, MeshGenFunc my_mg);
 
   // Re-used functionality in constructor
   void RegisterLoadBalancing_(ParameterInput *pin);
 
   void SetupMPIComms();
-};
-
-//----------------------------------------------------------------------------------------
-// \!fn Real ComputeMeshGeneratorX(std::int64_t index, std::int64_t nrange,
-//                                 bool sym_interval)
-// \brief wrapper fn to compute Real x logical location for either [0., 1.] or [-0.5, 0.5]
-//        real cell ranges for MeshGenerator_[] functions (default/user vs. uniform)
-
-inline Real ComputeMeshGeneratorX(std::int64_t index, std::int64_t nrange,
```

Comment:
The code path was used, but it was needlessly complex since the full functionality wasn't accessible anyway I think. I refactored what this stuff did to be much simpler and not have hard coded functions for each direction.